gui/backend/upsert_bot_config_test.go

```diff
@@ -0,0 +1,70 @@
+package backend
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stellar/kelp/gui/model2"
+	"github.com/stellar/kelp/support/kelpos"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsBotNameValid(t *testing.T) {
+	testCases := []struct {
+		name      string
+		botName   string
+		strategy  string
+		wantValid bool
+	}{
+		{
+			name:      "success default",
+			botName:   "George the Friendly Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success numeric",
+			botName:   "George the 1337 Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "success dash",
+			botName:   "George-the-Friendly-Octopus",
+			wantValid: true,
+		},
+		{
+			name:      "failure <>",
+			botName:   "George<>the Friendly Octopus",
```

Comment:
we should also test some other important invalid characters such as `.`, `&`, `_`, `(`, `)`, `+`, `@`